File: src/decorator/examples/caretakerrestrictions/NoMedicationRestriction.js
'use strict';

const CaretakerDecorator = require('./CaretakerDecorator');

class NoMedicationRestriction extends CaretakerDecorator {
  canAdministerMedication() { return false; }
  describe() { return `${super.describe()} | Rechaza administracion de medicamentos`; }
}

module.exports = NoMedicationRestriction;

File: src/decorator/examples/caretakerrestrictions/NoMedicationRestriction.java
package decorator.examples.caretakerrestrictions;

import factory.examples.caretakers.Caretaker;

public class NoMedicationRestriction extends CaretakerDecorator {

    public NoMedicationRestriction(Caretaker wrapped) { super(wrapped); }

    @Override public boolean canAdministerMedication() { return false; }

    @Override
    public String describe() {
        return super.describe() + " | Rechaza administracion de medicamentos";
    }
}
